.github/workflows/build.yml

```
@@ -0,0 +1,31 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build:
+    name: Build and analyze
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - uses: SonarSource/sonarqube-scan-action@v4
```

I had to look up what sonarqube is, might as well just add a comment to explain
```suggestion
      # SonarQube is an open-source platform for automated code quality and security reviews
      - uses: SonarSource/sonarqube-scan-action@v4
```